core/optaplanner-core-impl/pom.xml

```diff
@@ -30,6 +30,7 @@
     <dependency>
       <groupId>org.kie</groupId>
       <artifactId>kie-api</artifactId>
+      <optional>true</optional> <!-- Drools is only necessary together with DRL or CS-D. -->
     </dependency>
```

Comment:
We can not remove it completely, because score holders in API require this.
But they will only be classloaded when the DRL module is loaded, so the optional works well here.